draft-pda-core.md

```diff
@@ -291,11 +293,427 @@ Each PA task in this document is divided into three sub-protocols as follows.
 
 # PA protocols {#pa}
 
-[TODO(cjpatton)]
+This section specifies a protocol for executing generic PA tasks. Concrete
+PA protocols are described in {{prio}} and {{hits}}.
+
+Each round of the protocol corresponds to an HTTP request and response. The
+content type of each request is "application/octet-stream". We assume that some
+transport layer security protocol (e.g., TLS or QUIC) is used between each pair
+of parties and that the server is always authenticated.
+
+[TODO: Decide how to authenticate the leader in leader-to-helper and
+aggregator-to-collector connections. One option is to use client certificates
+for TLS; another is to have the leader sign its messages directly, as in Prio
+v2.]
+
+[TODO: @chris-wood suggested we specify APIs for producing and consuming each of
+the messages in the protocol. Specific PA protocols would implement this API.]
+
+[OPEN ISSUE: This needs to be reworked in order to account for protocols, like
+Hits, that involve multiple rounds of verify/collect. See issue #44.]
+
+**Error handling.**
+In this section, we will use the verbs "abort" and "alert with `[some error
+message]`" to describe how protocol participants react to various error
+conditions. The behavior is specified in {{pa-error}}. For common errors, we may
+elide the verbs altogether and refer to {{pa-error-common-aborts}}.
+
+## Configuration {#pa-config}
+
+### Tasks
+Each PA protocol is associated with a *PA task* that specifies the measurements
+that are to be collected and the protocol that will be used to collect them:
+
+~~~
+struct {
+  uint16 version;
+  opaque id[16];
+} PATask;
+
+~~~
+[TODO: Decide how to serialize protocol messages. We're using TLS syntax for
+now, but there's no reason to stick with it other than most folks are familiar
+with it.]
+
+The first field, `version` specifies the version of this document. The second
+field, `id` is an opaque identifier used by the clients, aggregators, and
+collector to uniquely identify the PA task at hand. We will call it the *task
+id* in the remainder.
+
+[TODO: Decide how the `PATask` is configured. Eventually this will be
+distributed, in an authenticated manner, from the collector the other parties.
+For now, we just assume this value is negotiated out-of-band.]
```

Comment:
I'm not quite following what's going on here. Why not just give each task version its own UID? Or alternatively have version somewhere else.

Reply:
Versioning the ID of a task seems fine, though not much different from each task ID being generated randomly. We can address this in a followup change. I'll clarify in the TODO.

Reply:
The version here is "version of the protocol." But maybe I'm misunderstanding you?